crates/driver/src/domain/quote.rs

```diff
@@ -88,13 +85,13 @@ impl Order {
         liquidity: &infra::liquidity::Fetcher,
         tokens: &infra::tokens::Fetcher,
     ) -> Result<Quote, Error> {
-        let liquidity = match solver.liquidity() {
-            solver::Liquidity::Fetch => {
+        let liquidity = match (solver.liquidity(), self.token_liquidity()) {
+            (solver::Liquidity::Fetch, pairs) if !pairs.is_empty() => {
                 liquidity
-                    .fetch(&self.liquidity_pairs(), infra::liquidity::AtBlock::Recent)
+                    .fetch(&pairs, infra::liquidity::AtBlock::Recent)
                     .await
             }
-            solver::Liquidity::Skip => Default::default(),
+            _ => Default::default(),
```

Comment:
You added the emptiness check in the `fetch` function. Here, it seems to be redundant now.
```suggestion
        let liquidity = match (solver.liquidity(), self.token_liquidity()) {
            (solver::Liquidity::Fetch, pairs) => {
liquidity
                    .fetch(&pairs, infra::liquidity::AtBlock::Recent)
.await
}
            _ => Default::default(),
```